compiler/rustc_codegen_gcc/build_system/src/config.rs

```diff
@@ -444,15 +444,17 @@ impl ConfigInfo {
             "build_sysroot/sysroot/lib/rustlib/{}/lib",
             self.target_triple,
         ));
-        let ld_library_path = format!(
-            "{target}:{sysroot}:{gcc_path}",
-            target = self.cargo_target_dir,
-            sysroot = sysroot.display(),
-            gcc_path = self.gcc_path,
-        );
-        env.insert("LIBRARY_PATH".to_string(), ld_library_path.clone());
-        env.insert("LD_LIBRARY_PATH".to_string(), ld_library_path.clone());
-        env.insert("DYLD_LIBRARY_PATH".to_string(), ld_library_path);
+        if !use_system_gcc {
+            let ld_library_path = format!(
+                "{target}:{sysroot}:{gcc_path}",
+                target = self.cargo_target_dir,
+                sysroot = sysroot.display(),
+                gcc_path = self.gcc_path,
+            );
+            env.insert("LIBRARY_PATH".to_string(), ld_library_path.clone());
+            env.insert("LD_LIBRARY_PATH".to_string(), ld_library_path.clone());
+            env.insert("DYLD_LIBRARY_PATH".to_string(), ld_library_path);
```

Comment:
So, at a glance, I won't claim to fully understand the number of paths that have been detected and variables that have been set by now, but should this propagate-forward the values of these variables that already exist on the system? Possibly as a secondary lookup? (so, suffixing the path?) Or is this only used in code that already does that?

Reply:
The build system of cg_gcc ignore a lot of things from the environment to have a controlled build, especially if it is embed into another build system (like the current one). I didn't add much flexibility into it yet. In this case, bootstrap passes `--use-system-gcc` and set `LD_LIBRARY_PATH` and don't want cg_gcc's build system to change it.

Reply:
Hokay!